scripts/exgfs_wave_post_gridded_sbs.sh

```diff
@@ -22,350 +22,140 @@
-
-# 1.a.1 Copy model definition files
-  for grdID in ${waveGRD} ${wavepostGRD} ${waveinterpGRD}; do
-    if [[ -f "${COMIN_WAVE_PREP}/${RUN}wave.mod_def.${grdID}" ]]; then
-      set +x
-      echo " Mod def file for ${grdID} found in ${COMIN_WAVE_PREP}. copying ...."
-      set_trace
-
-      cp -f "${COMIN_WAVE_PREP}/${RUN}wave.mod_def.${grdID}" "mod_def.${grdID}"
+DOGRI_WAV=${DOGRI_WAV:-"NO"}  # Interpolate to a grid
+DOGRB_WAV=${DOGRB_WAV:-"YES"} # Create grib2 files
+
+export waveGRD=${waveGRD?"FATAL ERROR: Required variable 'waveGRD' not set"}
+export waveinterpGRD=${waveinterpGRD?"FATAL ERROR: Required variable 'wavepostGRD' not set"}
+export wavepostGRD=${wavepostGRD?"FATAL ERROR: Required variable 'wavepostGRD' not set"}
+
+cat << EOF
+  INFO: Grid information:
+  INFO:   Native wave grids:  ${waveGRD}
+  INFO:   Interpolated grids: ${waveinterpGRD}
+  INFO:   Post-process grids: ${wavepostGRD}
+EOF
+
+fhr3=$(printf %03i ${FORECAST_HOUR})
+valid_time=$(date -u -d "${PDY} ${cyc} + ${FORECAST_HOUR} hours" "+%Y%m%d%H")
+
+cd "${DATA}" || exit 99
+
+# Copy model definition files
+for grdID in ${waveGRD} ${wavepostGRD} ${waveinterpGRD}; do
+  cpreq "${COMIN_WAVE_PREP}/${RUN}.wave.t${cyc}z.mod_def.${grdID}.bin" "mod_def.${grdID}"
+done
+
+# Copy model forecast data to DATA
+cpreq "${COMIN_WAVE_HISTORY}/${RUN}.wave.t${cyc}z.${waveGRD}.f${fhr3}.bin" "./out_grd.${waveGRD}"
+
+# Check for input templates for interpolation (copying will be done in the interpolation script)
+if [[ "${DOGRI_WAV}" = "YES" ]]; then
+  for intGRD in ${waveinterpGRD}; do
+    if [[ ! -f "${PARMgfs}/wave/${intGRD}_interp.inp.tmpl" ]]; then
+      echo "WARNING: No template for grid interpolation"
+      echo "WARNING: Interpolation will not be performed"
+      echo "WARNING: No products will be created for '${intGRD}'"
+      DOGRI_WAV="NO"
     fi
   done
-
-# 1.a.2 Check that model definition files exist
-  for grdID in ${waveGRD} ${wavepostGRD} ${waveinterpGRD}; do
-    if [[ ! -f "mod_def.${grdID}" ]]; then
-      set +x
-      echo ' '
-      echo '*************************************************** '
-      echo " FATAL ERROR : NO MOD_DEF FILE mod_def.$grdID "
-      echo '*************************************************** '
-      echo ' '
-      set_trace
-      err=2; export err;${errchk}
-      exit $err
-      DOGRB_WAV='NO'
-    else
-      set +x
-      echo "File mod_def.$grdID found. Syncing to all nodes ..."
-      set_trace
+fi
+
+# Check for input templates for grib2 products (copying will be done in the grib2 script)
+if [[ "${DOGRB_WAV}" = "YES" ]]; then
+  for grbGRD in ${waveinterpGRD} ${wavepostGRD}; do
+    if [[ ! -f "${PARMgfs}/wave/ww3_grib2.${grbGRD}.inp.tmpl" ]]; then
+      echo "FATAL ERROR: No template for grib generation"
+      DOGRB_WAV="NO"
     fi
   done
+fi
+
+# Data summary
+cat << EOF
+  INFO: Summary:
+  INFO:   Grid interp: DOGRI_WAV='${DOGRI_WAV}'
+  INFO:   Grib files:  DOGRB_WAV='${DOGRB_WAV}'
```

Comment:
`'${DOGRI_WAV}'` will be interpreted as a string, not an expansion of `${DOGRI_WAV}`.  Is that the intention?  If not, suggest double quotes:

```suggestion
  INFO:   Grid interp: DOGRI_WAV="${DOGRI_WAV}"
  INFO:   Grib files:  DOGRB_WAV="${DOGRB_WAV}"
```

Reply:
@aerorahul \*nudge\*

Reply:
done in 8d714478